text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,398 @@
+import logging
+import os
+from sklearn.model_selection import train_test_split
+from typing import Optional
+import numpy as np
+import json
+import pandas as pd
+
+from autogluon.core.utils import set_logger_verbosity
+from autogluon.core.utils.loaders import load_pd
+from autogluon.core.utils.utils import setup_outputdir, default_holdout_frac
+from autogluon.core.utils.miscs import in_ipynb
+
+from ..presets import ag_text_presets, merge_params
+from ..infer_types import infer_column_problem_types, printable_column_type_string
+from ..metrics import infer_eval_log_metrics
+from .. import constants as _C
+
+logger = logging.getLogger()  # return root logger
+
+
+class TextPredictor:
+    """AutoGluon TextPredictor predicts values in a column of a tabular dataset that contains texts
+    (classification or regression).
+
+    Parameters
+    ----------
+    label : str
+        Name of the column that contains the target variable to predict.
+    problem_type : str, default = None
+        Type of prediction problem, i.e. is this a binary/multiclass classification or regression
+        problem (options: 'binary', 'multiclass', 'regression').
+        If `problem_type = None`, the prediction problem type is inferred based on the
+        label-values in provided dataset.
+    eval_metric : function or str, default = None
+        Metric by which predictions will be ultimately evaluated on test data.
+        AutoGluon tunes factors such as hyper-parameters, early-stopping, ensemble-weights, etc.
+        in order to improve this metric on validation data.
+
+        If `eval_metric = None`, it is automatically chosen based on `problem_type`.
+        Defaults to 'roc_auc' for binary, 'accuracy' for multiclass classification, and
+        'root_mean_squared_error' for regression.
```

Comment:
standardize defaults across all tasks. I am fine with using 'roc_auc' as default for binary on all tasks as long as it's standardized across tasks.

Reply:
Lets set to accuracy for now, since vision doesn't support any metrics beyond accuracy (if I'm not mistaken), and because most models / automl packages default to accuracy outside of AG.

Reply:
Changed to accuracy.